pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -287,6 +287,13 @@ func (e *DockerRegistryServiceController) syncRegistryLocationChange() error {
 			if t.Type != v1.SecretTypeDockercfg {
 				continue
 			}
+			if t.Annotations == nil {
+				continue
+			}
+			// Do not manage dockercfg secrets we haven't created (eg. secrets created by user for private repositories).
+			if _, hasTokenSecret := t.Annotations[ServiceAccountTokenSecretNameKey]; !hasTokenSecret {
```

Comment:
It would be better to have something like "controlled-by" but I really don't want to deal with migration.

Reply:
(also I assume all secrets the controller create have this annotation)